.changeset/lucky-lizards-arrive.md

```diff
@@ -0,0 +1,6 @@
+---
+"@internal/fuel-core": patch
+"@fuel-ts/versions": patch
+---
+
+chore: upgrade `fuel-core` to `0.40.2`
```

Comment:
We can't merge this until the network is upgraded to this version. It's likely not happening in the next couple of weeks, so let's have this standby.

Reply:
@arboleya Yes I remember that after opening the PR. That is why I've made it a draft